catalog/transformer/catalog-transformer-shapefile/src/main/java/ddf/catalog/transformer/shape/file/transformer/ShapeFileInputTransformer.java

```diff
@@ -0,0 +1,120 @@
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.transformer.shape.file.transformer;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.types.Topic;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
+import ddf.catalog.transformer.shape.file.reader.api.ShapeFileHeader;
+import ddf.catalog.transformer.shape.file.reader.api.ShapeFileReader;
+import ddf.catalog.transformer.shape.file.reader.api.ShapeFileReaderFactory;
+import ddf.catalog.transformer.shape.file.transformer.api.ShapeFile;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashSet;
+import java.util.Set;
+
+/** This class is used convert the a .shp file into a metacard for DDF to display. */
+public class ShapeFileInputTransformer implements InputTransformer {
+
+  /** SHAPE_CATAGORY - catagory of the metacard. */
+  private static final String SHAPE_CATAGORY = "Shape File";
+
+  /** metacardType - the metacard type for the source. */
+  private MetacardType metacardType;
+
+  /** reader - Reads the contents of a shape file. */
+  private ShapeFileReaderFactory factory;
+
+  /**
+   * Constructor for ShapeFileInputTransformer class.
+   *
+   * @param type @see {@link
+   *     ddf.catalog.transformer.shape.file.transformer.ShapeFileInputTransformer#metacardType}
+   */
+  public ShapeFileInputTransformer(ShapeFileReaderFactory factory, MetacardType type) {
+    this.factory = factory;
+    this.metacardType = type;
+  }
+
+  /**
+   * Transforms an input stream into a metacard.
+   *
+   * @param input input stream containing the shape file
+   * @return Metacard the metacard created from the shape file
+   */
+  @Override
+  public Metacard transform(InputStream input) throws IOException, CatalogTransformerException {
+    return transform(input, null);
+  }
+
+  /**
+   * Transforms an input stream into a metacard and also allows for setting a custom ID for the
+   * metacard.
+   *
+   * @param input input stream containing the shape file
+   * @param id an id which can be used to set the metacard id
+   * @return Metacard the metacard created from the shape file
+   */
+  @Override
+  public Metacard transform(InputStream input, String id)
+      throws IOException, CatalogTransformerException {
+    return transformShapeFile(input, id);
+  }
+
+  /**
+   * @param input input stream containing the shape file
+   * @param id an id which can be used to set the metacard id
+   * @return the metacard created from the shape file
+   */
+  private Metacard transformShapeFile(InputStream input, String id)
+      throws CatalogTransformerException {
+    MetacardImpl metacard = new MetacardImpl(metacardType);
+
+    if (id != null) {
```

Comment:
✏️ nit, probably should just use `StinrgUtils.isNotBlank()` https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/StringUtils.html#isNotBlank(java.lang.String)